### PR TITLE
MINIFICPP-645 - Move from new to malloc in CAPI to facilitate eventua…

### PR DESCRIPTION
**Pull request by arpadboda**
…l change from C++ to C

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nanofi/src/api/nanofi.cpp [hunk 1]
```diff
@@ -205,7 +205,7 @@ flow_file_record* create_ff_object(const char *file, const size_t len, const uin
 }
 
 flow_file_record* create_ff_object_na(const char *file, const size_t len, const uint64_t size) {
-  flow_file_record *new_ff = new flow_file_record;
+  flow_file_record *new_ff = (flow_file_record*) malloc(sizeof(flow_file_record));
```
Comment (arpadboda) [inline nanofi/src/api/nanofi.cpp]: Happy newless year! :)

Comment (phrocker) [reply nanofi/src/api/nanofi.cpp]: Oops, sorry, must change the delete to free due to avoid heap corruption.

**Comment by arpadboda** [comment]
There is no delete, it was wrong before:
```
void free_flowfile(flow_file_record *ff) {
  if (ff == nullptr) {
    return;
  }
  auto content_repo_ptr = static_cast<std::shared_ptr<minifi::core::ContentRepository>*>(ff->crp);
  if (content_repo_ptr->get()) {
    std::shared_ptr<minifi::ResourceClaim> claim = std::make_shared<minifi::ResourceClaim>(ff->contentLocation, *content_repo_ptr);
    (*content_repo_ptr)->remove(claim);
  }
  if (ff->ffp == nullptr) {
    auto map = static_cast<string_map*>(ff->attributes);
    delete map;
  }
  free(ff->contentLocation);
  free(ff);
```

The last line is the one that frees.

**Comment by phrocker** [comment]
> There is no delete, it was wrong before, this PR just fixes:
> 
> ```
> void free_flowfile(flow_file_record *ff) {
>   if (ff == nullptr) {
>     return;
>   }
>   auto content_repo_ptr = static_cast<std::shared_ptr<minifi::core::ContentRepository>*>(ff->crp);
>   if (content_repo_ptr->get()) {
>     std::shared_ptr<minifi::ResourceClaim> claim = std::make_shared<minifi::ResourceClaim>(ff->contentLocation, *content_repo_ptr);
>     (*content_repo_ptr)->remove(claim);
>   }
>   if (ff->ffp == nullptr) {
>     auto map = static_cast<string_map*>(ff->attributes);
>     delete map;
>   }
>   free(ff->contentLocation);
>   free(ff);
> ```
> The last line is the one that frees.

Ah sorry, I was referencing the fact that over the course of PRs we've gone back and forth a little between malloc/new. There is a free_flow(flow *) that still uses delete. Happy to see a different PR if you prefer to do that, but it all falls under the guise of this ticket IMO. Would you prefer I merge this and then keep the ticket open as a blocker for the free? No real preference on my part.

**Pull request closed without merging**